quil-py/quil/program/__init__.pyi

```diff
@@ -175,6 +175,82 @@ class Program:
         while ensuring that unique value is not already in use within the program.
         """
         ...
+    def control_flow_graph(self) -> "ControlFlowGraph":
+        """
+        Return the `control flow graph`_ of the program.
+
+        .. _control flow graph: https://en.wikipedia.org/wiki/Control-flow_graph
+        """
+
+@final
+class BasicBlock:
+    def as_fixed_schedule(self, program: Program, include_zero_duration_instructions: bool) -> FixedSchedule:
+        """
+        Return the ``FixedSchedule`` representing the timing of the instructions within the block.
+
+        This schedule is computed by:
+
+        * Expanding each instruction within the block using the program's calibration definitions
+        * Resolving the `FixedSchedule` of the expanded instructions
+        * Mapping calibrated instructions back to the original instructions within this block, such that the
+          block's instruction is represented as a timespan encompassing all of its expanded instructions
+
+        If `include_zero_duration_instructions` is `True`, then the schedule will include instructions with zero duration
+        (such as `FENCE`). This may cause certain instructions to appear as unexpectedly long in duration.
```

Comment:
Why would zero-duration instructions cause "unexpectedly long" durations?

Reply:
I've been trying to think of how to illustrate this in the docs.

namely, on calibration expansion, one instruction is decomposed into many. Those decomposed instructions will now interact with other decomposed instructions in a way that "mix" the two original gates in time. A FENCE "plays" at the first clock tick following any preceding instructions on its qubits; that may not immediately precede the PULSE instruction whose timing the user actually cares about (for example).

Reply:
So zero-duration instructions can introduce new dependencies between previously-independent instructions?